packages/gatsby-theme-docs/src/templates/releases.js

```diff
@@ -3,147 +3,93 @@ import PropTypes from 'prop-types';
         <Markdown.TypographyPage>
           <SEO
-            title={props.pageContext.shortTitle || props.pageContext.title}
+            title={props.pageContext.title}
             excludeFromSearchIndex={props.pageContext.excludeFromSearchIndex}
           />
           <div>
-            <MDXRenderer>{props.data.releasesIndex.body}</MDXRenderer>
+            <MDXRenderer>{props.data.mdx.body}</MDXRenderer>
           </div>
           <div>
-            {props.data.releaseNotes.nodes.map((node, index) => {
-              return (
+            {props.data.allReleaseNotes &&
+              props.data.allReleaseNotes.nodes &&
+              props.data.allReleaseNotes.nodes.map(({ childMdx }, index) => (
                 <div key={index}>
-                  <Markdown.H2>{node.childMdx.fields.title}</Markdown.H2>
+                  <Markdown.H2>{childMdx.fields.title}</Markdown.H2>
                   <div>
-                    <MDXRenderer>{node.childMdx.body}</MDXRenderer>
+                    <MDXRenderer>{childMdx.body}</MDXRenderer>
                   </div>
-                  <hr />
                 </div>
-              );
-            })}
+              ))}
           </div>
         </Markdown.TypographyPage>
       </MDXProvider>
-    </LayoutContent>
+    </LayoutReleaseNotes>
   </ThemeProvider>
 );
 
-ReleaseNote.propTypes = {
+ReleaseNotesTemplate.propTypes = {
   pageContext: PropTypes.shape({
     slug: PropTypes.string.isRequired,
-    shortTitle: PropTypes.string,
     title: PropTypes.string.isRequired,
-    beta: PropTypes.bool.isRequired,
-    isGlobalBeta: PropTypes.bool.isRequired,
     excludeFromSearchIndex: PropTypes.bool.isRequired,
   }).isRequired,
   data: PropTypes.shape({
-    releasesIndex: PropTypes.shape({
+    mdx: PropTypes.shape({
       body: PropTypes.string.isRequired,
     }).isRequired,
-    releaseNotes: PropTypes.shape({
+    allReleaseNotes: PropTypes.shape({
       nodes: PropTypes.arrayOf(
         PropTypes.shape({
           childMdx: PropTypes.shape({
-            fields: PropTypes.shape({
-              title: PropTypes.string,
-            }),
+            title: PropTypes.string.isRequired,
             body: PropTypes.string.isRequired,
           }),
         })
-      ),
-    }).isRequired,
+      ).isRequired,
+    }),
   }).isRequired,
 };
 
-export default ReleaseNote;
+export default ReleaseNotesTemplate;
 
 export const query = graphql`
-  query QueryReleasesPage($slug: String!) {
-    releasesIndex: mdx(fields: { slug: { eq: $slug } }) {
+  query QueryReleasesPage($slug: String!, $isOverviewPage: Boolean!) {
+    mdx(fields: { slug: { eq: $slug } }) {
       body
     }
-
-    releaseNotes: allFile(filter: { sourceInstanceName: { eq: "releases" } }) {
+    allReleaseNotes: allFile(
+      filter: {
+        sourceInstanceName: { eq: "releases" }
+        internal: { mediaType: { eq: "text/mdx" } }
+        name: { ne: "index" }
+      }
+    ) @include(if: $isOverviewPage) {
```

Comment:
I added this so that this query is executed only when the page is the overview page.